### PR TITLE
Better error message for task.scope

### LINKED ISSUE #54178: `current_scope` is recommended by an error message but inaccessible.
This is not a great UX:

```julia
julia> t = Task(() -> 4)
Task (runnable) @0x0000ffff6826d370

julia> t.scope
ERROR: Querying `scope` is disallowed. Use `current_scope` instead.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] getproperty(t::Task, field::Symbol)
   @ Base ./task.jl:184
 [3] top-level scope
   @ REPL[65]:1

julia> current_scope
ERROR: UndefVarError: `current_scope` not defined in `Main`
Suggestion: check for spelling errors or missing imports.

julia> t.current_scope
ERROR: type Task has no field current_scope
Stacktrace:
 [1] getproperty(t::Task, field::Symbol)
   @ Base ./task.jl:186
 [2] top-level scope
   @ REPL[67]:1

julia> Base.current_scope
ERROR: UndefVarError: `current_scope` not defined in `Base`
Suggestion: check for spelling errors or missing imports.
Stacktrace:
 [1] getproperty(x::Module, f::Symbol)
   @ Base ./Base.jl:42
 [2] top-level scope
   @ REPL[68]:1

help?> current_scope
search: current_task current_exceptions

Couldn't find current_scope
Perhaps you meant current_task
  No documentation found.

  Binding current_scope does not exist.

julia> Core.current_scope
current_scope (built-in function)
```

https://github.com/JuliaLang/julia/blob/e57a7e49ad71241cc70ccdc0c1279274992f7dc9/base/task.jl#L183

Either the error should mention that the function is internal, or it should specify how to get a public handle on that function.

The error & hint was added in #52309 
cc @Keno

### PR DESCRIPTION
**Pull request by LilithHafner**
Fixes #54178

### CODE REVIEW

**Comment by LilithHafner** [comment]
Semantic merge conflict with #54176

File: base/task.jl [hunk 1]
```diff
@@ -180,7 +180,9 @@ end
         # TODO: this field name should be deprecated in 2.0
         return t._isexception ? t.result : nothing
     elseif field === :scope
-        error("Querying `scope` is disallowed. Use `current_scope` instead.")
+        error("""
+            Querying a Task's `scope` field is disallowed.
+            The private `Core.current_scope()` function is better, though still an implenetation detail.""")
```
Comment (Keno) [inline base/task.jl:185]: Typo

**Review by Keno — APPROVED** [review]
Fine by me

File: base/task.jl (hunk 1 above)
Comment (LilithHafner) [reply base/task.jl:185]:
```suggestion
            The private `Core.current_scope()` function is better, though still an implementation detail.""")
```

**Comment by LilithHafner** [comment]
Waiting for #54176 to merge.

**Pull request merged**